src/services/SplatalogueService.ts

```diff
@@ -67,139 +70,136 @@ export class SplatalogueService {
         this.axiosInstance = axios.create({baseURL: SplatalogueService.BaseUrl});
     }
 
-    async aliveCheck(): Promise<boolean> {
-        try {
-            const {status} = await this.axiosInstance.head("");
-            return status === 200;
-        } catch (err) {
-            return false;
-        }
-    }
-
-    async query(freqMin: number, freqMax: number, intensityLimit?: number): Promise<SpectralLineResponse> {
-        const url = SplatalogueService.ConstructUrl(freqMin, freqMax, intensityLimit);
-        const response = await this.axiosInstance.get(url);
-        return SplatalogueService.ConvertTable(response.data);
-    }
-
-    private static ConvertTable(tableString: string) {
-        if (!tableString || typeof tableString !== "string") {
-            throw new Error("invalid data received from Splatalogue");
-        }
+    query = async (freqMin: number, freqMax: number, intensityLimit?: number): Promise<SpectralLineResponse> => {
+        const params = SplatalogueService.GetParams(freqMin, freqMax, intensityLimit);
+        const response = await this.axiosInstance.post("", {body: JSON.stringify(params)});
+        return SplatalogueService.ConvertTable(response?.data);
+    };
 
-        const lines = tableString.split("\n");
-        if (!lines.length) {
+    private static ConvertTable = (data: object[]) => {
+        if (!data) {
             throw new Error("invalid data received from Splatalogue");
         }
 
-        const headerLine = lines[0];
-        const headerEntries = headerLine.split(":");
-        // First row is header, last row might be empty
-        let numDataRows = lines.length - 1;
-        if (lines[lines.length - 1] === "") {
-            numDataRows--;
-        }
-
-        const numColumns = headerEntries.length;
-
-        if (numColumns !== SplatalogueService.SplatalogueHeaders.length) {
-            throw new Error("Unexpected header data received from Splatalogue");
-        }
+        // remove data with invalid species id and invalid species name
+        // this will remove data which only stores the query string
+        data = data.filter(line => line?.["species_id"] !== null || line?.["name"]);
```

Comment:
If line is undefined, then undefined !== null would equal true. Is that what you want? Also, should the || be && instead according to your comment on line 84?